amivapi/validation.py

```diff
@@ -121,3 +122,22 @@ def _validate_filetype(self, filetype, field, value):
         if not(t in filetype):
             self._error(field, "filetype '%s' not supported, has to be in: "
                         "%s" % (t, filetype))
+
+    def _validate_session_younger_than(self, threshold_time, field, value):
+        """Validation of the used token for special fields
+
+        Validates if the session is not older than threshold_time
+
+        Except admins, they can ignore this
+
+        Args:
+            threshold_time (int): threshold to to compare with in seconds
+            field (string): field name.
+            value: field value.
+        """
+        time_created = g.current_session['_created'].replace(tzinfo=None)
```

Comment:
you should put an `if not g.get('resource_admin'):` around everything so this really only applies to non-admins.